ece595_project_2/source/app_supervisor.c

```diff
@@ -208,3 +211,9 @@ void Create_Periodic_OS_Tasks(void)
         }
     }
 }
+
+static void Detect_Pet_Task(void *pvParameters)
+{
+	Detect_Pet();
+	vTaskDelay(pdMS_TO_TICKS(1000));
```

Comment:
1 second is probably too slow. If you want to get multiple detections as the pet passes through, it should probably be 200ms or so.